src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs

```diff
@@ -0,0 +1,98 @@
+        }
+
+        internal static IVsProjectRestoreInfo Build(IEnumerable<IProjectVersionedValue<IProjectSubscriptionUpdate>> updates)
+        {
+            Requires.NotNull(updates, nameof(updates));
+
+            // if none of the underlying subscriptions have any changes
+            if (!updates.Any(u => u.Value.ProjectChanges.Any(c => c.Value.Difference.AnyChanges)))
+            {
+                return null;
+            }
+
+            string baseIntermediatePath = null;
+            var targetFrameworks = new TargetFrameworks();
+            
+            foreach (IProjectVersionedValue<IProjectSubscriptionUpdate> update in updates)
+            {
+                var configurationChanges = update.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+                baseIntermediatePath = baseIntermediatePath ?? 
+                    configurationChanges.After.Properties[ConfigurationGeneral.BaseIntermediateOutputPathProperty];
+                string targetFrameworkMoniker = 
+                    configurationChanges.After.Properties[ConfigurationGeneral.TargetFrameworkMonikerProperty];
+
+                if (targetFrameworks.Item(targetFrameworkMoniker) == null)
+                {
+                    var projectReferencesChanges = update.Value.ProjectChanges[ProjectReference.SchemaName];
+                    var packageReferencesChanges = update.Value.ProjectChanges[PackageReference.SchemaName];
+
+                    targetFrameworks.Add(new TargetFrameworkInfo
+                    {
+                        TargetFrameworkMoniker = targetFrameworkMoniker,
+                        ProjectReferences = GetProjectReferences(projectReferencesChanges.After.Items),
+                        PackageReferences = GetReferences(packageReferencesChanges.After.Items)
+                    });
+                }
+            }
+
+            return new ProjectRestoreInfo
+            {
+                BaseIntermediatePath = baseIntermediatePath,
+                TargetFrameworks = targetFrameworks
+            };
+        }
+
+        private static IVsReferenceItems GetReferences(IImmutableDictionary<String, IImmutableDictionary<String, String>> items)
+        {
+            return new ReferenceItems(items.Select(p => new ReferenceItem
+            {
+                Name = p.Key,
+                Properties = new ReferenceProperties(p.Value.Select(v => new ReferenceProperty
+                {
+                    Name = v.Key, Value = v.Value
+                })) 
+            }));
+        }
+
+        private static IVsReferenceItems GetProjectReferences(IImmutableDictionary<String, IImmutableDictionary<String, String>> items)
+        {
+            var referenceItems = GetReferences(items);
+            foreach (ReferenceItem item in referenceItems)
+            {
+                var definingProjectDirectory = item.Properties.Item(DefiningProjectDirectoryProperty);
```

Comment:
Oh Identity isn't a resolved path - we use UnconfiguredProject.MakeRooted everywhere else. We should at least be consistent. I can see how UnconfiguredProject.MakeRooted might be out of sync. I don' think Path.GetFullPath is correct though - as that will use the current working directory of VS to figure out the path.